### PR TITLE
capcom/lwings.cpp: hooked up MCU for avengers [David Haywood]

### PR DESCRIPTION
**Pull request by Osso13**
If/when this gets merged I (Ivan) will probably do a round of cleanups (i.e. moving avengers in a subclass since it's the only set in the driver that needs the MCU emulation code). Just didn't want to make it more difficult to check the actual changes for this PR.

### CODE REVIEW

File: src/mame/capcom/lwings.cpp [hunk 1]
```diff
@@ -121,161 +125,23 @@ void lwings_state::avengers_interrupt(int state)
...
 	{
-		m_param[3] = data;
-	}
-	else if (pc == 0x0445)
-	{
-		m_soundstate = 0x80;
-		m_soundlatch->write(data);
+		m_mcu->set_input_line(0, ASSERT_LINE);
+		m_irq_ack_timer->adjust(attotime::from_msec(1000/60));
 	}
```
Comment (cuavas) [inline src/mame/capcom/lwings.cpp:134]: You probably want to either use a latch device, or use a sync barrier so the microcontroller sees the interrupt and updated value at the same time.

File: src/mame/capcom/lwings.cpp [hunk 2]
```diff
@@ -1062,15 +932,83 @@ void lwings_state::trojan(machine_config &config)
...
+	// possibly 0xbf -> 0xff transition - NO can't be, 0xbf doesn't happen on param writes for angle?
+	//if (data == 0xff)
+	//  m_mcu->set_input_line(0, CLEAR_LINE);
+}
+
+
+TIMER_CALLBACK_MEMBER(lwings_state::irq_ack)
+{
+	m_mcu->set_input_line(0, CLEAR_LINE);
+}
```
Comment (cuavas) [inline src/mame/capcom/lwings.cpp:986]: This seems a bit hacky.  What does the actual interrupt handler code look like?

File: src/mame/capcom/lwings.cpp [hunk 3]
```diff
@@ -1062,15 +932,83 @@ void lwings_state::trojan(machine_config &config)
...
+void lwings_state::mcu_p1_w(u8 data)
+{
+}
+
+void lwings_state::mcu_p2_w(u8 data)
+{
+	m_mcu->set_input_line(0, CLEAR_LINE); // not sure, we also clear on a timed ack
+	m_soundstate = 0x80;
+	m_soundlatch->write(data);
+}
```
Comment (cuavas) [inline src/mame/capcom/lwings.cpp:972]: The I/O ports aren’t strobed, so clearing the interrupt on any write isn’t possible.

**Comment by ajrhacker** [comment]
I've put together a somewhat less hacky version of these changes. I've avoided setting bizarre and inexplicable CPU clock speeds by doing something a bit more clever (but intensive).

**Pull request closed without merging**